src/Core/Services/Implementations/UserService.cs

```diff
@@ -1355,13 +1361,27 @@ public void SetTwoFactorProvider(User user, TwoFactorProviderType type, bool set
     private async Task CheckPoliciesOnTwoFactorRemovalAsync(User user)
     {
         var twoFactorPolicies = await _policyService.GetPoliciesApplicableToUserAsync(user.Id, PolicyType.TwoFactorAuthentication);
+        var organizationsManagingUser = await GetOrganizationsManagingUserAsync(user.Id);
 
         var removeOrgUserTasks = twoFactorPolicies.Select(async p =>
         {
-            await _removeOrganizationUserCommand.RemoveUserAsync(p.OrganizationId, user.Id);
             var organization = await _organizationRepository.GetByIdAsync(p.OrganizationId);
-            await _mailService.SendOrganizationUserRemovedForPolicyTwoStepEmailAsync(
-                organization.DisplayName(), user.Email);
+            if (_featureService.IsEnabled(FeatureFlagKeys.AccountDeprovisioning) && organizationsManagingUser.Any(o => o.Id == p.OrganizationId))
```

Comment:
I was thinking of moving organizationsManagingUser into the if statement so it's only called if the feature check is on, but your way is cleaner, and if performance isn't an issue, I think it's the better option.

Reply:
If the feature flag is disabled then `GetOrganizationsManagingUserAsync` simply returns an empty list.
ref: https://github.com/bitwarden/server/blob/170836aba1f87ee4a5c5a0b33d76dfa487ee94d1/src/Core/Services/Implementations/UserService.cs#L1295